### ISSUE TITLE
Core 2024.11b0 prevents integration from loading

### ISSUE THREAD

**Issue by TekFan** [custom-components/remote_homeassistant#321]
Just upgraded to core 2024.11b0 and remote_homeassistant integration doesn't load anymore.

Here are the errors in the log:

Logger: homeassistant.setup
Source: setup.py:334
First occurred: 10:46:15 (1 occurrences)
Last logged: 10:46:15

Setup failed for custom integration 'remote_homeassistant': Unable to import component: cannot import name 'DATA_CUSTOMIZE' from 'homeassistant.config' (/usr/src/homeassistant/homeassistant/config.py)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1011, in async_get_component
    comp = await self.hass.async_add_import_executor_job(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1071, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/remote_homeassistant/__init__.py", line 22, in <module>
    from homeassistant.config import DATA_CUSTOMIZE
ImportError: cannot import name 'DATA_CUSTOMIZE' from 'homeassistant.config' (/usr/src/homeassistant/homeassistant/config.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component
    component = await integration.async_get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1031, in async_get_component
    self._component_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1023, in async_get_component
    comp = self._get_component()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1071, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/remote_homeassistant/__init__.py", line 22, in <module>
    from homeassistant.config import DATA_CUSTOMIZE
ImportError: cannot import name 'DATA_CUSTOMIZE' from 'homeassistant.config' (/usr/src/homeassistant/homeassistant/config.py)

**Comment by chasebrizy9** [comment]
Same exact issue here, as stated above.

**Comment by TekFan** [comment]
New additional error since Core 2024.11b1:

Logger: homeassistant.util.loop
Source: util/loop.py:77
First occurred: 12:44:31 (1 occurrences)
Last logged: 12:44:31

Detected blocking call to import_module with args ('custom_components.remote_homeassistant',) in /usr/src/homeassistant/homeassistant/loader.py, line 1071: ComponentProtocol, importlib.import_module(self.pkg_path) inside the event loop; This is causing stability issues. Please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#import_module Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 189, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete self.run_forever() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever self._run_once() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once handle._run() File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/setup.py", line 165, in async_setup_component result = await _async_setup_component(hass, domain, config) File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component component = await integration.async_get_component() File "/usr/src/homeassistant/homeassistant/loader.py", line 1023, in async_get_component comp = self._get_component() File "/usr/src/homeassistant/homeassistant/loader.py", line 1071, in _get_component ComponentProtocol, importlib.import_module(self.pkg_path)

**Comment by Plawasan** [comment]
`/remote_homeassistant/__init__.py`

```
22 - from homeassistant.config import DATA_CUSTOMIZE
22 + from homeassistant.core_config import DATA_CUSTOMIZE
```
Resolved the issue for me

**Comment by Bluhme1** [comment]
Great. Thank you. Works for me.

**Comment by TekFan** [comment]
Yep, much better, thanks !

I'll leave this issue open atm to remind the dev to add this modification.

**Comment by bartbakels** [comment]
In the release of 2034.11 the integration does not start..

**Comment by PacmanForever** [comment]
> In the release of 2034.11 the integration does not start..

One solution here: https://github.com/custom-components/remote_homeassistant/issues/321#issuecomment-2452426910

**Comment by jaym25** [comment]
Suggest using...

`22 - from homeassistant.config import DATA_CUSTOMIZE`
`22 + from homeassistant.helpers.entity import DATA_CUSTOMIZE`

which is backwards compatible

**Comment by snigehere** [comment]
> [home-assistant/core#129065](https://github.com/home-assistant/core/pull/129065)
> 
> `/homeassistant/custom_components/remote_homeassistant/__init__.py`
> 
> ```
> 22 - from homeassistant.config import DATA_CUSTOMIZE
> 22 + from homeassistant.core_config import DATA_CUSTOMIZE
> ```
> 
> Resolved the issue for me

Many thanks for posting this.. it solved the issue for me.

**Comment by r0bb10** [comment]
> Suggest using...
> 
> `22 - from homeassistant.config import DATA_CUSTOMIZE` `22 + from homeassistant.helpers.entity import DATA_CUSTOMIZE`
> 
> which is backwards compatible

confirm this solution works! great!

**Comment by lukas-hetzenecker** [comment]
Fixed with #323
